packages/flutter/lib/src/painting/image_decoder.dart

```diff
@@ -24,7 +24,16 @@ import 'binding.dart';
 /// in tests.
 Future<ui.Image> decodeImageFromList(Uint8List bytes) async {
   final ui.ImmutableBuffer buffer = await ui.ImmutableBuffer.fromUint8List(bytes);
-  final ui.Codec codec = await PaintingBinding.instance.instantiateImageCodecWithSize(buffer);
-  final ui.FrameInfo frameInfo = await codec.getNextFrame();
-  return frameInfo.image;
+  try {
+    final ui.Codec codec = await PaintingBinding.instance.instantiateImageCodecWithSize(buffer);
+    final ui.FrameInfo frameInfo;
+    try {
+      frameInfo = await codec.getNextFrame();
```

Comment:
Could you just put `return await codec.getNextFrame();` here?

Reply:
Thank you. [9a63a86](https://github.com/flutter/flutter/pull/161488/commits/9a63a8693ad796171006b0b1cd8dc7a34f6d6f08)